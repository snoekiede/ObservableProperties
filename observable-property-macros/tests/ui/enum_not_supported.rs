use observable_property::Observable;

// This should fail to compile because Observable can only be derived for structs
#[derive(Observable)]
enum BadEnum {
    Variant1,
    Variant2,
}

fn main() {}
