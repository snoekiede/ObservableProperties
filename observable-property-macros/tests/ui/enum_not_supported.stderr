error: Observable can only be derived for structs
 --> tests/ui/enum_not_supported.rs:5:1
  |
5 | / enum BadEnum {
6 | |     Variant1,
7 | |     Variant2,
8 | | }
  | |_^
